firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteDocumentOverlayCache.java

```diff
@@ -59,13 +59,16 @@ public Mutation getOverlay(DocumentKey key) {
   }
 
   private void saveOverlay(int largestBatchId, DocumentKey key, @Nullable Mutation mutation) {
-    String collectionPath = EncodedPath.encode(key.getPath().popLast());
+    ResourcePath collectionPath = key.getPath().popLast();
+    String group = collectionPath.getLastSegment();
```

Comment:
Can we use `key.getCollectionGroup()` here?

Reply:
Oh, right!